### DISCUSSION TITLE
How to exit a shell script using gum

### DISCUSSION THREAD

**Discussion by jbmoorhouse** [charmbracelet/gum#351] [Q&A] [answered]
## Problem
Firstly, thank you producing such a great tool!

I'm having issues quitting a program using gum.

## To Reproduce

Given the following shell script. If I press `CTRL + C`, this only quits the current gum step. i.e. in the example below, I have to `CTRL + C` 3 times.

```bash
gum confirm "foo"
gum confirm "bar"
gum confirm "baz"
```

## Expected behavior
More of a question, is this expected? Is there a way to exit the program entirely?

## Setup Info
OS: Ubuntu 20.04.3 LTS
gum version:  v0.10.0

**Comment by caarlos0** [comment, marked as answer]
Hey @jbmoorhouse!

Yes, this is bound to happen, as `gum` is another process within the script.

You can propagate it by adding a `set -e`, e.g.:

```bash
#!/bin/bash
set -e 
gum confirm "foo"
gum confirm "bar"
gum confirm "baz"
```

That means that if anything inside the script has an exit code > 0, it will fail right away, so pressing CTRL+C (or replying "no") in any of the confirms quits the entire script propagating the exit code. All this to say that, in your script, if you don't wan't the script to on, say, replying "no", you'll need to handle the exit codes.

For reference, `gum confirm`:
- `exit 0` on "Yes"
- `exit 1` on "No"
- `exit 130` on CTRL+C

Here's an example handling the exit codes:

```bash
#!/bin/bash
set -e

if gum confirm "Do you want some soda?" ; then
	if gum confirm "What about some chips with that soda?" ; then
		gum format "OK, here's your soda with chips!"
		exit 0
	elif test $? -eq 130; then
		exit 130
	else
		gum format "OK, here's your soda!"
		exit 0
	fi
elif test $? -eq 130; then
	exit 130
else
	if gum confirm "OK, no soda, what about some coffee?"; then
		gum format "Cool, here's your coffee!"
	elif test $? -eq 130; then
		exit 130
	else
		gum format "OK, just water then, incoming!"
		exit 0
	fi
fi
```

Hope that helps :)

**Reply by jbmoorhouse** [reply]
Perfect, just what I needed. Thank you very much, closing

**Reply by mylinuxforwork** [reply]
Is the exit 130 still working with the latest version? The script above is not working anymore to handle 130 exit code.

**Comment by cr2007** [comment]
Does this work even for `gum input` as well?

Like when I try to assign a value to a variable via `gum input`,
and then I try to exit the program via <kbd>Ctrl</kbd> + <kbd>C</kbd>, it does not exit the program, and just runs.

<img width="200" alt="image" src="https://github.com/charmbracelet/gum/assets/73425927/4a116662-9106-4f12-90f6-40608dda411d">
